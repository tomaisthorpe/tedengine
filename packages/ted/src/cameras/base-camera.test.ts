import { vec3 } from 'gl-matrix';
import { TBaseCamera } from '../index';

test('moveBy should move by given amount', () => {
  const camera = new TBaseCamera();

  camera.moveBy(vec3.fromValues(1, 2, 3));
  expect([...camera.cameraComponent.transform.translation]).toEqual([1, 2, 3]);

  // Run again to ensure isn't setting the values
  camera.moveBy(vec3.fromValues(2, 4, 6));
  expect([...camera.cameraComponent.transform.translation]).toEqual([3, 6, 9]);
});

test('moveTo should set to the given position', () => {
  const camera = new TBaseCamera();

  camera.moveTo(vec3.fromValues(1, 2, 3));
  expect([...camera.cameraComponent.transform.translation]).toEqual([1, 2, 3]);

  // Run again to ensure isn't adding values
  camera.moveTo(vec3.fromValues(2, 4, 6));
  expect([...camera.cameraComponent.transform.translation]).toEqual([2, 4, 6]);
});

test('onUpdate should call update on controller', async () => {
  const camera = new TBaseCamera();
  const controller = { onUpdate: jest.fn() };

  camera.controller = controller;

  await camera.onUpdate({} as any, 0);

  expect(controller.onUpdate).toHaveBeenCalledTimes(1);
});

test('lookAt should set the rotation correctly', () => {
  const camera = new TBaseCamera();

  camera.lookAt(vec3.fromValues(1, 2, 3));
  expect([...camera.cameraComponent.transform.rotation]).toMatchSnapshot();

  // Run again with different values
  camera.lookAt(vec3.fromValues(-1, -2, -3));
  expect([...camera.cameraComponent.transform.rotation]).toMatchSnapshot();
});
