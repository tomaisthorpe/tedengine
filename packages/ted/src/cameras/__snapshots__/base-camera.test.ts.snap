// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TBaseCamera lookAt should set the rotation correctly 1`] = `
[
  -0.04457065090537071,
  0.9481061697006226,
  -0.274656742811203,
  -0.15385644137859344,
]
`;

exports[`TBaseCamera lookAt should set the rotation correctly 2`] = `
[
  -0.274656742811203,
  0.15385644137859344,
  0.04457065090537071,
  0.9481061697006226,
]
`;
